src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1509,6 +1539,10 @@ public override BoundNode VisitLocal(BoundLocal node)
             }
 
             SetResult(node, GetAdjustedResult(type, slot), type);
+            if (_reinferredVariablesOpt?.GetReinferredSymbol(node.LocalSymbol) is LocalSymbol updated)
+            {
+                SetUpdatedSymbol(node, node.LocalSymbol, updated);
```

Comment:
>node [](start = 33, length = 4)

The mapping does not depend on the node, we shouldn't be caching per node. And shouldn't be doing this on every VisitLocal.

Reply:
As discussed offline, it does for method calls, properties, and anything else where generics can be reinferred.

---
In reply to: [327882499](https://github.com/dotnet/roslyn/pull/38769#discussion_r327882499) [](ancestors = 327882499)